src/rgw/rgw_op.cc

```diff
@@ -5284,6 +5284,62 @@ int RGWCopyObj::verify_permission(optional_yield y)
 					      RGW_PERM_READ)) { 
 	  return -EACCES;
 	}
+
+  // Check for tagging permissions
+  auto tagging_drctv = s->info.env->get("HTTP_X_AMZ_TAGGING_DIRECTIVE");
+  TaggingDirective itag_der = TaggingDirective::NO_TAGGING_DIRECTIVE;
+  if (tagging_drctv) {
+    if (strcmp(tagging_drctv, "COPY") == 0)
```

Comment:
use strncmp() here and elsewhere code is added, to check directive more stringently along with its length

Reply:
I don't want to see any use of C string functions in any new code, to be honest.

Reply:
@mattbenjamin I have now used string compare() function. Is that okay?